pkg/oonimkall/taskrunner_test.go

```diff
@@ -354,77 +370,15 @@ func TestTaskRunnerRun(t *testing.T) {
 		assertReducedEventsLike(t, expect, reduced)
 	})
 
-	t.Run("with missing input and InputOrQueryBackend policy", func(t *testing.T) {
+	t.Run("with error during target loading", func(t *testing.T) {
 		runner, emitter := newRunnerForTesting()
 		fake := fakeSuccessfulDeps()
-		fake.Builder.MockInputPolicy = func() model.InputPolicy {
-			return model.InputOrQueryBackend
-		}
-		runner.newSession = fake.NewSession
-		events := runAndCollect(runner, emitter)
-		reduced := reduceEventsKeysIgnoreLog(t, events)
-		expect := []eventKeyCount{
-			{Key: eventTypeStatusQueued, Count: 1},
-			{Key: eventTypeStatusStarted, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 3},
-			{Key: eventTypeStatusGeoIPLookup, Count: 1},
-			{Key: eventTypeStatusResolverLookup, Count: 1},
-			{Key: eventTypeFailureStartup, Count: 1},
-			{Key: eventTypeStatusEnd, Count: 1},
-		}
-		assertReducedEventsLike(t, expect, reduced)
-	})
-
-	t.Run("with missing input and InputStrictlyRequired policy", func(t *testing.T) {
-		runner, emitter := newRunnerForTesting()
-		fake := fakeSuccessfulDeps()
-		fake.Builder.MockInputPolicy = func() model.InputPolicy {
-			return model.InputStrictlyRequired
-		}
-		runner.newSession = fake.NewSession
-		events := runAndCollect(runner, emitter)
-		reduced := reduceEventsKeysIgnoreLog(t, events)
-		expect := []eventKeyCount{
-			{Key: eventTypeStatusQueued, Count: 1},
-			{Key: eventTypeStatusStarted, Count: 1},
-			{Key: eventTypeStatusProgress, Count: 3},
-			{Key: eventTypeStatusGeoIPLookup, Count: 1},
-			{Key: eventTypeStatusResolverLookup, Count: 1},
-			{Key: eventTypeFailureStartup, Count: 1},
-			{Key: eventTypeStatusEnd, Count: 1},
-		}
-		assertReducedEventsLike(t, expect, reduced)
-	})
-
-	t.Run("with InputOrStaticDefault policy and experiment with no static input",
-		func(t *testing.T) {
-			runner, emitter := newRunnerForTesting()
-			runner.settings.Name = "Antani" // no input for this experiment
-			fake := fakeSuccessfulDeps()
-			fake.Builder.MockInputPolicy = func() model.InputPolicy {
-				return model.InputOrStaticDefault
-			}
-			runner.newSession = fake.NewSession
-			events := runAndCollect(runner, emitter)
-			reduced := reduceEventsKeysIgnoreLog(t, events)
-			expect := []eventKeyCount{
-				{Key: eventTypeStatusQueued, Count: 1},
-				{Key: eventTypeStatusStarted, Count: 1},
-				{Key: eventTypeStatusProgress, Count: 3},
-				{Key: eventTypeStatusGeoIPLookup, Count: 1},
-				{Key: eventTypeStatusResolverLookup, Count: 1},
-				{Key: eventTypeFailureStartup, Count: 1},
-				{Key: eventTypeStatusEnd, Count: 1},
+		fake.Builder.MockNewTargetLoader = func(config *model.ExperimentTargetLoaderConfig) model.ExperimentTargetLoader {
+			return &mocks.ExperimentTargetLoader{
+				MockLoad: func(ctx context.Context) ([]model.ExperimentTarget, error) {
+					return nil, errors.New("mocked error")
+				},
```

Comment:
Because now we don't switch explicitly on the InputPolicy, we need just one test case for failing target loading.